File: TTTLinkPK/LinkPK/PKManager.m
//
//  PKManager.m
//  LinkPK
//
//  Created by yanzhen on 2019/1/28.
//  Copyright © 2019 yanzhen. All rights reserved.
//

#import "PKManager.h"

@implementation PKManager
static id _manager;
+ (instancetype)manager
{
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _manager = [[self alloc] init];
    });
    return _manager;
}

+(instancetype)allocWithZone:(struct _NSZone *)zone
{
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _manager = [super allocWithZone:zone];
    });
    return _manager;
}

- (id)copyWithZone:(NSZone *)zone
{
    return _manager;
}

- (instancetype)init
{
    self = [super init];
    if (self) {
        //初始化TTTRtcEngineKit对象，输入申请的AppID
        _rtcEngine = [TTTRtcEngineKit sharedEngineWithAppId:<#AppId#> delegate:nil];
    }
    return self;
}
@end

File: TTTLinkPK-Swift/LinkPK/PKManager.swift
//
//  PKManager.swift
//  LinkPK
//
//  Created by Work on 2019/3/14.
//  Copyright © 2019 yanzhen. All rights reserved.
//

import UIKit
import TTTRtcEngineKit

class PKManager: NSObject {

    public static let manager = PKManager()
    public var rtcEngine: TTTRtcEngineKit!
    public var roomID: Int64 = 0 //频道id
    public var uid: Int64 = 0    //自己id
    private override init() {
        super.init()
        //初始化TTTRtcEngineKit对象，输入申请的AppID
        rtcEngine = TTTRtcEngineKit.sharedEngine(withAppId: <#AppId#>, delegate: nil)
    }
}
